beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperIntegrationTestBase.java

```diff
@@ -0,0 +1,240 @@
+
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.core.model.HousekeepingPath;
+import com.expediagroup.beekeeper.core.model.LifecycleEventType;
+import com.expediagroup.beekeeper.integration.utils.ContainerTestUtils;
+import com.expediagroup.beekeeper.integration.utils.MySqlTestUtils;
+
+@Testcontainers
+public abstract class BeekeeperIntegrationTestBase {
+
+  // AWS VARIABLES
+  private static final String AWS_ACCESS_KEY_ID = "access-key";
+  private static final String AWS_SECRET_KEY = "secret-key";
+
+  // BEEKEEPER DB VARIABLES
+  private static final String BEEKEEPER_DB_NAME = "beekeeper";
+  private static final String BEEKEEPER_FLYWAY_TABLE = "flyway_schema_history";
+  private static final String BEEKEEPER_HOUSEKEEPING_PATH_TABLE_NAME = "housekeeping_path";
+  private static final String BEEKEEPER_HOUSEKEEPING_METADATA_TABLE_NAME = "housekeeping_metadata";
+
+  // FIELDS TO INSERT INTO BEEKEEPER TABLES
+  private Long id = 1L;
+  private static final String HOUSEKEEPING_PATH_FIELDS = String.join(",", ID_FIELD, PATH_FIELD, DATABASE_NAME_FIELD,
+      TABLE_NAME_FIELD, HOUSEKEEPING_STATUS_FIELD, CREATION_TIMESTAMP_FIELD, MODIFIED_TIMESTAMP_FIELD,
+      CLEANUP_TIMESTAMP_FIELD, CLEANUP_DELAY_FIELD, CLEANUP_ATTEMPTS_FIELD, CLIENT_ID_FIELD, LIFECYCLE_TYPE_FIELD);
+  private static final String HOUSEKEEPING_METADATA_FIELDS = String.join(",", ID_FIELD, PATH_FIELD, DATABASE_NAME_FIELD,
+      TABLE_NAME_FIELD, PARTITION_NAME_FIELD, HOUSEKEEPING_STATUS_FIELD, CREATION_TIMESTAMP_FIELD,
+      MODIFIED_TIMESTAMP_FIELD, CLEANUP_TIMESTAMP_FIELD, CLEANUP_DELAY_FIELD, CLEANUP_ATTEMPTS_FIELD, CLIENT_ID_FIELD,
+      LIFECYCLE_TYPE_FIELD);
+  private static final String LIFE_CYCLE_FILTER = "WHERE " + LIFECYCLE_TYPE_FIELD + " = '%s' ORDER BY " + PATH_FIELD;
+  private static final String LIFE_CYCLE_AND_UPDATE_FILTER = "WHERE " + LIFECYCLE_TYPE_FIELD + " = '%s'"
+      + " AND " + MODIFIED_TIMESTAMP_FIELD + " > " + CREATION_TIMESTAMP_FIELD
+      + " ORDER BY " + PATH_FIELD;
+
+  // MySQL DB CONTAINER AND UTILS
+  @Container
+  private static final MySQLContainer MY_SQL_CONTAINER = ContainerTestUtils.mySqlContainer();
+  private static MySqlTestUtils mySQLTestUtils;
+
+  protected final ExecutorService executorService = Executors.newFixedThreadPool(1);
+
+  @BeforeAll
+  protected static void initMySQLContainer() throws SQLException {
+    String jdbcUrl = MY_SQL_CONTAINER.getJdbcUrl() + "?useSSL=false";
+    String username = MY_SQL_CONTAINER.getUsername();
+    String password = MY_SQL_CONTAINER.getPassword();
+
+    System.setProperty("spring.datasource.url", jdbcUrl);
+    System.setProperty("spring.datasource.username", username);
+    System.setProperty("spring.datasource.password", password);
+    System.setProperty("aws.region", AWS_REGION);
+    System.setProperty("aws.accessKeyId", AWS_ACCESS_KEY_ID);
+    System.setProperty("aws.secretKey", AWS_SECRET_KEY);
```

Comment:
Same as my other comment, we often use https://stefanbirkner.github.io/system-rules/ to manage anything to do with the `System` calls so it might be useful here.